transmart-api-server/src/main/groovy/org/transmartproject/api/server/client/CustomRestTemplate.groovy

```diff
@@ -0,0 +1,11 @@
+package org.transmartproject.api.server.client
+
+import org.springframework.web.client.RestOperations
+import org.springframework.web.client.RestTemplate
+
+class CustomRestTemplate extends RestTemplate implements RestOperations {
```

Comment:
We don't have to have a custom class. Couldn't it be matter composing in the spring configuration like below:
```
@Bean
RestTemplate offlineTokenBasedRestTemplate(CustomClientRequestFactory factory) {
 new RestTemplate(factory)
}
```